### PR TITLE
Graph custom colors

### PR DESCRIPTION
**Pull request by krzysztofruszczynski**
This is:

```
- [ ] a bugfix
- [x] a new feature
```

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [ ] Documentation is updated as necessary

### Why this change is needed?
To customize colors on pie and donut charts.

### CODE REVIEW

File: src/PhpSpreadsheet/Chart/DataSeriesValues.php [hunk 1]
```diff
@@ -60,9 +60,9 @@ class DataSeriesValues
     private $dataValues = [];
 
     /**
-     * Fill color.
+     * Fill color (can be array with colors if dataseries have custom colors).
      *
-     * @var string
+     * @var array|string
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Chart/DataSeriesValues.php]: This should be and all similar type hinting too:

```suggestion
     * @var string[]|string
```

File: src/PhpSpreadsheet/Chart/DataSeriesValues.php [hunk 2]
```diff
@@ -224,14 +224,22 @@ public function getFillColor()
...
-        if (!preg_match('/^[a-f0-9]{6}$/i', $color)) {
-            throw new Exception('Invalid hex color for chart series');
+        if (is_array($color)) {
+            foreach ($color as $colorValue) {
+                if (!preg_match('/^[a-f0-9]{6}$/i', $colorValue)) {
+                    throw new Exception(sprintf('Invalid hex color for chart series (color: "%s")', $colorValue));
+                }
+            }
+        } else {
+            if (!preg_match('/^[a-f0-9]{6}$/i', $color)) {
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Chart/DataSeriesValues.php]: Avoid duplicating validation code by creating a private method like "validateColor(string $color): void" that throw exception if need be.

File: src/PhpSpreadsheet/Writer/Xlsx/Chart.php [hunk 3]
```diff
@@ -1097,24 +1097,48 @@ private function writePlotGroup($plotGroup, $groupType, $objWriter, &$catIsMulti
...
+
             if (($groupType == DataSeries::TYPE_PIECHART) || ($groupType == DataSeries::TYPE_PIECHART_3D) || ($groupType == DataSeries::TYPE_DONUTCHART)) {
-                $objWriter->startElement('c:dPt');
-                $objWriter->startElement('c:idx');
-                $objWriter->writeAttribute('val', 3);
-                $objWriter->endElement();
+                $fillColorValues = $plotSeriesValues->getFillColor();
+                if ($fillColorValues !== null && is_array($fillColorValues)) {
+                    foreach ($plotSeriesValues->getDataValues() as $dataKey => $dataValue) {
+                        $objWriter->startElement('c:dPt');
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Writer/Xlsx/Chart.php]: This should be extract into private method to avoid duplicating code

File: src/PhpSpreadsheet/Writer/Xlsx/Chart.php [hunk 4]
```diff
@@ -1127,9 +1151,6 @@ private function writePlotGroup($plotGroup, $groupType, $objWriter, &$catIsMulti
                 $objWriter->endElement();
             }
 
-            //    Values
-            $plotSeriesValues = $plotGroup->getPlotValuesByIndex($plotSeriesRef);
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Writer/Xlsx/Chart.php]: Why was this removed ?

Comment (krzysztofruszczynski) [reply src/PhpSpreadsheet/Writer/Xlsx/Chart.php]: It was moved to line 1100 ($plotSeriesValues is needed earlier).

**Comment by PowerKiKi** [comment]
Thanks, I rebased and merged your work.
